iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tools/cleanup/InfinispanTenantCleaner.java

```diff
@@ -96,59 +131,61 @@ public void run() throws Exception {
 
                 // delete device connection entry first
 
-                final DeviceConnectionKey conKey = new DeviceConnectionKey(tenantIdToClean, deviceId);
-                this.devicesConnectionCache.remove(conKey);
+                if (devicesConnectionCache != null) {
+                    final DeviceConnectionKey conKey = new DeviceConnectionKey(tenantIdToClean, deviceId);
+                    devicesConnectionCache.remove(conKey);
+                }
 
                 // delete device registration second
 
                 final DeviceKey regKey = DeviceKey
                         .deviceKey(TenantHandle.of(tenantIdToClean, tenantIdToClean), deviceId);
-                this.devicesCache.remove(regKey);
+                devicesCache.remove(regKey);
 
             });
 
-            log.info("Deleted {} entries", matches.size());
+            count += matches.size();
+
+            log.info("Deleted {} entries in this iteration", matches.size());
 
         } while (matches.size() > 0);
```

Comment:
This result in an extra hotrod request + server query which may be costly. Testing `matches.size()=DELETION_CHUNK_SIZE` avoid that last call